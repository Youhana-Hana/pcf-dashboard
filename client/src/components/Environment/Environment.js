import React, { Component } from 'react';

import {
  Row,
  Col,
  Card,
  CardHeader,
  CardBlock,
  CardFooter,
  CardTitle,
  Badge
} from "reactstrap";

import CardRow from '../CardRow/';

class Environment extends Component {

  getStatusLabel(status) {
    switch(status.toLowerCase()) {
      case "passed":
        return "S";
      case "failed":
        return "F";
      default:
        return "R";
    }
  }

  getStatusColor(status) {
    switch(status.toLowerCase()) {
      case "passed":
        return "success";
      case "failed":
        return "danger";
      default:
        return "warning";
    }
  }


  render() {
    let url = this.props.env.buildInfo.buildUrl;
    let buildNumber = this.props.env.buildInfo.buildGlobalIdentifier;
    let failedJobName = this.props.env.buildInfo.failedJobName;

    return (
      <Col xs="12" sm="6" lg="4">
        <CardHeader>
          <h2 className="mb-0">
            {this.props.env.region} {this.props.env.env}
            <Badge pill color={this.getStatusColor(this.props.env.pipelineStatus)} className="float-right"><a href={url} className="text-white" target="_blank"> {buildNumber}</a></Badge>
          </h2>
        </CardHeader>

        <Card className="text-white bg-primary">
          <CardBlock className="card-body pb-0">
            <CardRow row= {{label:"Current ERT Version", value: this.props.env.currentErtVersion}} key={this.props.env.currentErtVersion}></CardRow>

            <CardRow row= {{label:"Forthcoming ERT Version(S3)", value: this.props.env.latestErtVersionInS3}} key={this.props.env.latestErtVersionInS3}></CardRow>

            <CardRow row= {{label:"PCF Pipeline Version", value: this.props.env.pcfPipelineVersion}} key={this.props.env.pcfPipelineVersion}></CardRow>

            {failedJobName ?
             <CardRow row= {{label:"Failed Job", value: failedJobName, url: url }} key={failedJobName}></CardRow> : ""}
          </CardBlock>
        </Card>
      </Col>
    )
  }
}

export default Environment;
