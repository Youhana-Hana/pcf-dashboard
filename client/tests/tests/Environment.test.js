import React from 'react';
import renderer from 'react-test-renderer';
import Environment  from '../../src/components/Environment/';


const environment = {
  "id": "us-t4",
  "foundation": "T4",
  "region": "US",
  "products": {
    "name": "elastic-runtime",
    "currentVersion": "1.10.13",
    "stagedVersion": "",
    "currentVersionInS3": "1.10.17",
    "pcfPipelineVersion": "1.15.1",
    "pipelineStatus": "failed",
    "buildInfo": {
      "pipelineName": "upgrade-tile-ert",
      "buildNumber": "22",
      "buildGlobalIdentifier": "56",
      "buildUrl": "http://myconcourse-on-T4us.com/builds/55",
      "failedJobName": "configure-ert"
    }
  }
};

it('test to see if the environment renders correctly', () => {
  const environmentView = renderer
  .create(<Environment env={environment} />)
  .toJSON();
  expect(environmentView).toMatchSnapshot();
});
