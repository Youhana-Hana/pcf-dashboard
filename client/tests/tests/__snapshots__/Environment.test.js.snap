// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ERT Version only 1`] = `
<div
  className="col-12 col-sm-8 col-lg-4"
>
  <div
    className="card-header"
  >
    <h2
      className="mb-0"
    >
       
    </h2>
  </div>
  <div
    className="text-white bg-primary card"
  >
    <div
      className="card-body pb-0 card-block"
    >
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Current ERT Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            1.10.13
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Staged ERT Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            1.11.23
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Current OpsManager Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            TBC
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Staged OpsManager Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            N/A
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`test to see if the environment renders correctly 1`] = `
<div
  className="col-12 col-sm-8 col-lg-4"
>
  <div
    className="card-header"
  >
    <h2
      className="mb-0"
    >
       
    </h2>
  </div>
  <div
    className="text-white bg-primary card"
  >
    <div
      className="card-body pb-0 card-block"
    >
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Current ERT Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            1.10.13
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Staged ERT Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            N/A
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Current OpsManager Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            1.11.11
          </h3>
        </div>
      </div>
      <div
        className="padding-10 row"
      >
        <div
          className="col-12 col-sm-6 col-lg-8"
        >
          <h3>
            Staged OpsManager Version
          </h3>
        </div>
        <div
          className="col-10 col-sm-6 col-lg-4"
        >
          <h3>
            1.12.12
          </h3>
        </div>
      </div>
    </div>
  </div>
</div>
`;
